man/sd_pidfd_get_inode_id.xml

```diff
@@ -0,0 +1,107 @@
+    <refpurpose>Acquire the 64-bit inode id of a PID file descriptor (PIDFD)</refpurpose>
+  </refnamediv>
+
+  <refsynopsisdiv>
+    <funcsynopsis>
+      <funcsynopsisinfo>#include &lt;systemd/sd-daemon.h&gt;</funcsynopsisinfo>
+
+      <funcprototype>
+        <funcdef>int <function>sd_pidfd_get_inode_id</function></funcdef>
+        <paramdef>int <parameter>pidfd</parameter></paramdef>
+        <paramdef>uint64_t *<parameter>ret</parameter></paramdef>
+      </funcprototype>
+    </funcsynopsis>
+  </refsynopsisdiv>
+
+  <refsect1>
+    <title>Description</title>
+
+    <para><function>sd_pidfd_get_inode_id()</function> may be invoked to acquire the 64-bit inode id of
+    a PID file descriptor (PIDFD), which can be used to reliably identify a process.</para>
+
+    <para>As a typical example, the service manager sets <varname>$MAINPIDFDID</varname> and <varname>$MANAGERPIDFDID</varname>
+    environment variables to the inode ids of the service main process and the service manager itself, respectively,
+    if such functionality is supported by the kernel.</para>
+
+    <para>On 64-bit architectures, the inode id can be directly obtained via a call to
+    <citerefentry project='man-pages'><refentrytitle>fstat</refentrytitle><manvolnum>2</manvolnum></citerefentry>
+    on a given pidfd. However, on 32-bit architectures the <structname>struct stat</structname>'s .st_ino
+    field is also 32-bit, which similar to PIDs is subject to reuse. Therefore, a second mechanism leveraging
+    <citerefentry project='man-pages'><refentrytitle>name_to_handle_at</refentrytitle><manvolnum>2</manvolnum></citerefentry>
+    has been added to kernel. This helper is added to simplify downstream handling of pidfd/pidfs internals.
+    </para>
+  </refsect1>
+
+  <refsect1>
+    <title>Return Value</title>
+
+    <para>On success, the function returns 0 or a positive integer. On failure, a negative errno-style
+    error code is returned.</para>
+
+    <refsect2>
+      <title>Errors</title>
+
+      <para>Returned errors may indicate the following problems:</para>
+
+      <variablelist>
+
+        <varlistentry>
+          <term><constant>-EOPNOTSUPP</constant></term>
+
+          <listitem><para>The stable PIDFD inode id is not supported by the running kernel, or the system
+          is 32-bit and <function>name_to_handle_at()</function> is unavailable.</para></listitem>
+        </varlistentry>
+
+        <varlistentry>
+          <term><constant>-EBADF</constant></term>
+
+          <listitem><para>The specified file descriptor is invalid, or is not a PIDFD.</para></listitem>
+        </varlistentry>
+
+      </variablelist>
+    </refsect2>
+  </refsect1>
+
+  <refsect1>
+    <title>Notes</title>
+
+    <xi:include href="libsystemd-pkgconfig.xml" xpointer="pkgconfig-text"/>
+  </refsect1>
+
+  <refsect1>
+    <title>History</title>
+    <para><function>sd_pidfd_get_inode_id()</function> was added in version 258.</para>
```

Comment:
can you also add this to NEWS right-away. Now that we have it mostly set up for v258 please make sure to keep it up-to-date